Sources/Parameters.swift

```diff
@@ -6,16 +6,34 @@
 
 import Foundation
 
-public enum ParametersError: Error {
-  case invalidSyntax(value: Any)
-  case invalidKey(key: String, value: Any)
-  case invalidStructure(key: String, oldValue: Any, newValue: Any)
-}
+// For retro-compatibility. Remove in next major.
+@available(*, deprecated, renamed: "Parameters.Error", message: "Use the Parameters.Error nested type instead")
+public typealias ParametersError = Parameters.Error
 
+/// Namespace to handle extra context parameters passed as a list of `foo=bar` strings.
+/// Typically used when parsing command-line arguments one by one
+/// (like `foo=bar pt.x=1 pt.y=2 values=1 values=2 values=3 flag`)
+/// to turn them into a dictionary structure
 public enum Parameters {
+  public enum Error: Swift.Error {
+    case invalidSyntax(value: Any)
+    case invalidKey(key: String, value: Any)
+    case invalidStructure(key: String, oldValue: Any, newValue: Any)
+  }
+
   typealias Parameter = (key: String, value: Any)
   public typealias StringDict = [String: Any]
 
+  /// Transforms a list of strings representing structred-key/value pairs, like
+  /// `["pt.x=1", "pt.y=2", "values=1", "values=2", "values=3" "flag"]`
+  /// into a structured dictionary.
+  ///
+  /// - Parameter items: The list of `k=v`-style Strings, each string
```

Comment:
Well when you ask Xcode to generate the doc comment (menu "Editor" > "Structure" > "Add Documentation"), it uses the "`- Parameters` + one param per line" if there's more than one parameter, but uses directly `- Parameter foo` if there's only one parameter.

I agree that's inconsistent, and I actually wondered the same as you here. idk if it's better to force consistency ourselves, using `- Parameters` + one param per line, or to just do as Xcode does (which at least would not force us to fight Xcode and correct every new contributor who are using Xcode's doc auto-generation shortcut)